packages/kbn-apm-synthtrace-client/src/lib/logs/index.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { Fields } from '../entity';
+import { Serializable } from '../serializable';
+
+export type LogDocument = Fields &
+  Partial<{
+    'input.type': string;
+    'log.file.path'?: string;
+    'service.name': string;
+    'data_stream.namespace': string;
+    'data_stream.type': string;
+    'data_stream.dataset': string;
+    message?: string;
+    'event.dataset': string;
+    'log.level'?: string;
+    'host.name'?: string;
+  }>;
+
+class Log extends Serializable<LogDocument> {
+  service(name: string) {
+    this.fields['service.name'] = name;
+    return this;
+  }
+
+  namespace(value: string) {
+    this.fields['data_stream.namespace'] = value;
+    return this;
+  }
+
+  dataset(value: string) {
+    this.fields['data_stream.dataset'] = value;
+    this.fields['event.dataset'] = value;
+    return this;
+  }
+
+  logLevel(level: string) {
+    this.fields['log.level'] = level;
+    return this;
+  }
+
+  message(message: string) {
+    this.fields.message = message;
+    return this;
+  }
+}
+
+function service(serviceName: string = 'synth-service'): Log {
+  return new Log({
```

Comment:
`service.name` is optional for logs, does it make sense to structure the scenarios around this?

Reply:
Good point, we might also have infra logs. I will remove and make it generic

Reply:
Side note: We should push customers to annotate logs with service.name. For application logs we should make it required.

Reply:
My understanding is this is for the UI testing so it must work with `service.name` and without it.

Reply:
Sure, just saying I think `service.name` so be very central to log ingestion (also our tooling)